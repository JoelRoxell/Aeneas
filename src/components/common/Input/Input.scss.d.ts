// This file is generated automatically
export const input: string;
export const label: string;
export const textField: string;
export const err: string;
export const success: string;
