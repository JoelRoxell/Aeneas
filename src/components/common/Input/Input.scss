.input {
  .text-field {
    width: 100%;
    outline: none;
    box-shadow: none;
    border: none;
    border-bottom: rem(1) solid #ccc;
    height: rem(30);
    font-size: rem(12);
    transition: color 0.3s, border 0.3s;
    appearance: none;
    outline: none;
    margin-bottom: rem(30);
    background: transparent;

    &:focus {
      box-shadow: none;
      border-bottom-color: $active-color;
      color: $active-color;
    }
  }
}
