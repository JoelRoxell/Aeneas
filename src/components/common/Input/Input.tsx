import * as React from 'react'
import * as style from './Input.scss'

interface InputProps {
  title: string
  value: string | number
  placeholder: string
  autoComplete: string
  name?: string
  type?: string
  onChange?: (inputName: string, value: string) => any
}

function Input(props: InputProps) {
  return (
    <div className={style.input}>
      <div className={style.label}>{props.title}</div>

      <input
        className={style.textField}
        name={props.name || props.title}
        type={props.type || 'text'}
        value={props.value}
        placeholder={props.placeholder}
        autoComplete={props.autoComplete}
        onChange={event => {
          if (props.onChange) {
            props.onChange(event.target.name, event.target.value)
          }
        }}
      />
    </div>
  )
}

export default Input
