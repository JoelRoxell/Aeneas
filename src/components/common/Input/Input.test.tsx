import * as React from 'react'
import { mount, shallow } from 'enzyme'
import Input from 'components/common/Input'
import { act } from 'react-dom/test-utils'

const InputWrapper = () => {
  const [text, setText] = React.useState('')

  return (
    <Input
      autoComplete="email"
      placeholder="email"
      title="email"
      value={text}
      onChange={(_, value) => {
        setText(value)
      }}
    />
  )
}

describe('Input', () => {
  test('render', () => {
    let wrapper = shallow(
      <Input
        autoComplete="email"
        placeholder="email"
        title="email"
        value={'default-text'}
        onChange={(_, __) => {}}
        leftIcon="left"
        rightIcon="right"
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('change', () => {
    let wrapper = mount(<InputWrapper />)

    act(() => {
      wrapper
        .find('input')
        .simulate('change', { target: { name: 'input', value: 'Some input text' } })
    })
    wrapper.update()
    expect(wrapper.find('input').prop('value')).toEqual('Some input text')
  })
})
