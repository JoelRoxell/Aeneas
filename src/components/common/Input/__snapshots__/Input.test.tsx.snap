// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render 1`] = `
<div
  className="input"
>
  <div
    className="label"
  >
    email
  </div>
  <div
    className="inputWrapper"
  >
    <Icon
      className="leftIcon"
      height={24}
      name="left"
      width={24}
    />
    <input
      autoComplete="email"
      className="textField"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="email"
      type="text"
      value="default-text"
    />
    <Icon
      className="rightIcon"
      height={24}
      name="right"
      width={24}
    />
  </div>
</div>
`;
